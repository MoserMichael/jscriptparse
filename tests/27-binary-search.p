tosearch=[2, 4, 5, 7, 9, 10, 12, 14, 15, 17]


println("searching:", tosearch)

def search(tosearch, findme) {
  low=0
  high=len(tosearch)-1

  while low <= high {
    middle = int( (high + low)/2 )

    if tosearch[ middle ] == findme 
        return true
    elif tosearch[ middle ] > findme 
        high = middle-1
     else 
        low = middle+1
    
  }
  return false
}

println("10 ", search(tosearch,10))
println("11 ", search(tosearch,11))
println("12 ", search(tosearch,12))
println("13 ", search(tosearch,13))
println("14 ", search(tosearch,14))

