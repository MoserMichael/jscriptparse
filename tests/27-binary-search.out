searching: [2,4,5,7,9,10,12,14,15,17]
10  true
11  false
12  true
13  false
14  true
